app/models/lookup_value.rb

```diff
@@ -37,6 +38,11 @@ def value=(val)
     end
   end
 
+  def hidden_value=(val)
+    lookup_key.hidden_value = val
+    lookup_key.save unless lookup_key.new_record?
```

Comment:
c. lookup_key is saved in the DB, so it *should* already be valid. there is no validation on the hidden_value field on it, so this shouldn't be a problem...